### ISSUE TITLE
cover does not work when actual percentage 0

### ISSUE THREAD

**Issue by nh2** [nick8325/quickcheck#26]
If you write with `cover`: "I want that x% (e.g. 100%) of my generated characters are `a`", and no `a`s are generated at all, `cover` doesn't warn you when it should.

Example 1:

```haskell
quickCheck $ forAll (return 'b') $ \c -> cover (c == 'a') 100 "a" True
+++ OK, passed 100 tests.
```

The problem is that quickcheck only considers `stamp`s that actually occur.

This also happens if the percentage is very small so that with high probability no suitable sample is found; example 2:

```haskell
> quickCheck $ forAll arbitrary $ \c -> cover (c == 'a') 100 "a" True
+++ OK, passed 100 tests (only 1% a; not 100%).
> quickCheck $ forAll arbitrary $ \c -> cover (c == 'a') 100 "a" True
+++ OK, passed 100 tests.
```

The location where `cover` is handled is `covers` in `success :: State -> IO ()` in `QuickCheck.Test`.

**Comment by nick8325** [comment]
Thanks for the bug report, this should be fixed now!
